### PR TITLE
more macro hygiene cleanup: test #[pymethods] and more arg parsing and protos

### PR DESCRIPTION
**Pull request by birkenfeld**
After this PR and the two previous ones (#1841, #1842), no non-doc usages of `pyo3::` remain in the macros-backend crate or the main crate.

### CODE REVIEW

File: tests/test_proc_macro_hygiene.rs [hunk 1]
```diff
@@ -26,6 +26,43 @@ pub struct Bar {
...
+    #[classmethod]
+    fn clsmethod(_: &::pyo3::types::PyType) {}
+    #[call]
+    #[args(args = "*", kwds = "**")]
+    fn __call__(
+        &self,
+        _args: &::pyo3::types::PyTuple,
+        _kwds: ::std::option::Option<&::pyo3::types::PyDict>,
+    ) -> ::pyo3::PyResult<i32> {
+        ::std::unimplemented!()
```
Comment (mejrs) [inline tests/test_proc_macro_hygiene.rs:44]:
```suggestion
        ::std::panic!("unimplemented isn't hygienic before 1.50")
```

File: tests/test_proc_macro_hygiene.rs [hunk 2]
```diff
@@ -43,6 +80,19 @@ impl ::pyo3::class::gc::PyGCProtocol for Bar {
...
 
+#[cfg(not(Py_LIMITED_API))]
+#[::pyo3::proc_macro::pyproto]
+impl ::pyo3::class::PyBufferProtocol for Bar {
+    fn bf_getbuffer(
+        _s: ::pyo3::PyRefMut<Self>,
+        _v: *mut ::pyo3::ffi::Py_buffer,
+        _f: ::std::os::raw::c_int,
+    ) -> ::pyo3::PyResult<()> {
+        ::std::unimplemented!()
```
Comment (mejrs) [inline tests/test_proc_macro_hygiene.rs:91]:
```suggestion
        ::std::panic!("unimplemented isn't hygienic before 1.50")
```

**Review by mejrs — CHANGES_REQUESTED** [review]
Looks like unimplemented isn't hygienic before 1.50 - https://github.com/rust-lang/rust/pull/78343

**Comment by birkenfeld** [comment]
> Looks like unimplemented isn't hygienic before 1.50 - [rust-lang/rust#78343](https://github.com/rust-lang/rust/pull/78343)

Interesting, thanks for finding that :)

**Review by davidhewitt — APPROVED** [review]
💯 awesome work on building out this test @birkenfeld @mejrs !

**Pull request merged**